subsys/pm/pm.c

```diff
@@ -176,6 +179,9 @@ bool pm_system_suspend(int32_t ticks)
 
 	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, ticks);
 
+	/* Exiting active state */
+	pm_state_notify(PM_STATE_EXIT);
+
```

Comment:
At this point you don't know yet that you will go to any of the lower power states. Notification seems to be to early.